src/ucp/core/ucp_worker.c

```diff
@@ -535,18 +536,23 @@ ucs_status_t ucp_worker_set_ep_failed(ucp_worker_h worker, ucp_ep_h ucp_ep,
 
     if ((ucp_ep_ext_gen(ucp_ep)->err_cb == NULL) &&
         (ucp_ep->flags & UCP_EP_FLAG_USED)) {
+        /* do not print error if connection reset by remote peer since it can
+         * be part of user level close protocol  */
```

Comment:
what if it's in case of an unexpected (not user intentional) disconnect? how would the user be notified of an error?

Reply:
that bothered me as well, but if the user wants to avoid processing errors, then it's his ~problem~decision

Reply:
> what if it's in case of an unexpected (not user intentional) disconnect? how would the user be notified of an error?

we can't detect if it's expected or not disconnect event, user should set error callback to be aware about any type of errors and handle it properly.